src/MaxMind/Db/Reader.php

```diff
@@ -237,8 +238,10 @@ private function findMetadataStart($filename)
         $fileSize = $fstat['size'];
         $marker = self::$METADATA_START_MARKER;
         $markerLength = self::$METADATA_START_MARKER_LENGTH;
+        $maxMetadataSize = self::$METADATA_MAX_SIZE;
+        // $maxMetadataSize = $fileSize - $markerLength;
 
-        for ($i = 0; $i < $fileSize - $markerLength + 1; $i++) {
+        for ($i = 0; $i < min($maxMetadataSize, $fileSize - $markerLength) + 1; $i++) {
```

Comment:
It might be good to move `min($maxMetadataSize, $fileSize - $markerLength) + 1` for clarity and so that it doesn't have to be recomputed each iteration.